<?xml version="1.0" ?>
<launch>

     <include file="$(find astra_camera)/launch/astra.launch"/>

     <node pkg="target_finder" type="target_locator_srv" name="target_locator" output="screen" >
          <param name="camera_name" value="/camera/rgb"/>
          <param name="image_topic" value="/camera/rgb/image_rect_color" />
          <param name="target_type" value="2" />
          <param name="target_rows" value="12" />
          <param name="target_cols" value="9" />
          <param name="target_circle_dia" value="0.0125" />
          <param name="target_spacing" value="0.020" />
          <param name="focal_length_x" value="570.3422241210938" />
          <param name="focal_length_y" value="570.3422241210938"/>
          <param name="center_x" value="319.5" />
          <param name="center_y" value="239.5" />
          <param name="use_circle_detector" value="true"/>
     </node>
 
     <node pkg="target_finder" type="call_service" name="call_service" output="screen" >
           <param name="roi_width" value="640"/>
           <param name="roi_height" value="480"/>
           <param name="optical_frame" value="camera_rgb_optical_frame"/>
     </node>
 
     <node pkg="image_view" type="image_view" name="debug_view" >
          <remap from="image" to="observer_raw_image"/>
     </node>

     <node pkg="image_view" type="image_view" name="rect_view" >
          <remap from="image" to="/camera/rgb/image_rect_color"/>
     </node>

     <node pkg="tf" type="static_transform_publisher" name="target_frame" args="0.1 0.0 1.0 0.0 1.5708 0.0 world camera_rgb_optical_frame 100" />
     <node name="rviz" pkg="rviz" type="rviz" args="-d $(find target_finder)/config/target_finder_astra.rviz" required="true" />

</launch>
